apisix/stream/xrpc/runner.lua

```diff
@@ -71,9 +79,58 @@ local function put_req_ctx(session, ctx)
 end
 
 
+local function filter_logger(ctx, logger)
+    if not logger then
+       return false
+    end
+
+    if not logger.filter or #logger.filter == 0 then
+        -- no valid filter, default execution plugin
+        return true
+    end
+
+    -- key and version are divided by per-logger level
+    local key = "xrpc-logger" .. ctx.conf_id
+    local version = tostring(logger.filter)
```

Comment:
Actually, we can use the key directly as the version so we can save a `tostring`.
As each logger's conf's memory address is independent.

Reply:
But the meaning of `version` here should be related to `filter`. If `conf` does not change but `filter` changed, then `version` should change and a new expr should be generated?

However, as currently designed, if the `conf` is changed via the admin api, the `filter` will also change and use the new memory address.

I was trying to make the place more intuitive.